dcos_launch/terraform.py

```diff
@@ -0,0 +1,144 @@
+import json
+import os
+import re
+import shutil
+import subprocess
+
+import dcos_launch.config
+import yaml
+from dcos_launch import gcp, util
+
+
+class TerraformLauncher(util.AbstractLauncher):
+    def __init__(self, config: dict, env=None):
+        self.config = config
+        self.init_dir = dcos_launch.config.expand_path('', self.config['init_dir'])
+        self.cluster_profile_path = os.path.join(self.init_dir, 'desired_cluster_profile.tfvars')
+
+        if config['platform'] == 'azure':
+            # TODO
+            pass
+        elif config['platform'] == 'aws':
+            # TODO
+            pass
+        elif config['platform'] == 'gcp':
+            creds_string, creds_path = gcp.get_credentials(env)
+            if not creds_path:
+                creds_path = os.path.abspath('gcp_service_creds.json')
+                with open(creds_path, 'w') as file:
+                    file.write(creds_string)
+            config['terraform_config']['gcp_credentials_key_file'] = creds_path
+            if 'gcp_project' not in config['terraform_config']:
+                config['terraform_config']['gcp_project'] = json.loads(creds_string)['project_id']
+        else:
+            raise util.LauncherError('InvalidPlatform', "allowed values: 'gcp', 'aws' and 'azure'")
+
+    def create(self):
+        # create terraform directory
+        if os.path.exists(self.init_dir):
+            raise util.LauncherError('ClusterAlreadyExists', "Either the cluster you are trying to create is already "
+                                                             "running or the init_dir you specified in your config is "
+                                                             "already used by another active cluster.")
+        os.makedirs(self.init_dir)
+        module = 'github.com/dcos/{}?ref={}/{}'.format(
+            'terraform-dcos-enterprise' if self.config['dcos-enterprise'] else 'terraform-dcos',
+            self.config['terraform_dcos_enterprise_version'] if self.config['dcos-enterprise'] else
+            self.config['terraform_dcos_version'], self.config['platform'])
+
+        """ Converting our YAML config to the required format. You can find an example of that format in the Advance
```

Comment:
only use docs string notation like this under method declarations. Use `#` here

Reply:
Done